Makefile

```diff
@@ -90,11 +90,21 @@ manager: $(GO)
 container: manager
 	$(OCI_BIN) build build/ -t ${REGISTRY}/${IMG}:${IMAGE_TAG}
 
+check-tag-exists:
+	@echo "Checking if tag '${IMAGE_GIT_TAG}' exists in ${REGISTRY}/${IMG}..."
+	@if skopeo inspect docker://${REGISTRY}/${IMG}:${IMAGE_GIT_TAG} >/dev/null 2>&1; then \
+	    echo "Tag '${IMAGE_GIT_TAG}' already exists. Skipping push."; \
+	    exit 0; \
+	fi
+
 # Push the docker image
 docker-push:
 	$(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_TAG}
-	$(OCI_BIN) tag ${REGISTRY}/${IMG}:${IMAGE_TAG} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}
-	$(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}
+	@$(MAKE) check-tag-exists || \
+	( \
+	    $(OCI_BIN) tag ${REGISTRY}/${IMG}:${IMAGE_TAG} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}; \
+	    $(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}; \
+	)
```

Comment:
Can it be simplified please to one target ? (something like this, but didn't check it is correct, just as a direction)

```
docker-push:
	@echo "Checking if tag '${IMAGE_GIT_TAG}' exists in ${REGISTRY}/${IMG}..."
	@if ! skopeo inspect docker://${REGISTRY}/${IMG}:${IMAGE_GIT_TAG} >/dev/null 2>&1; then \
	    echo "Tag '${IMAGE_GIT_TAG}' does not exist. Pushing image..."; \
	    $(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_TAG}; \
	    $(OCI_BIN) tag ${REGISTRY}/${IMG}:${IMAGE_TAG} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}; \
	    $(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}; \
	else \
	    echo "Tag '${IMAGE_GIT_TAG}' already exists. Skipping push."; \
	fi
```

Reply:
sure, DONE

Reply:
[DONE](https://github.com/k8snetworkplumbingwg/kubemacpool/compare/0c5cb46f61336203023119ea6b58945a4f09d9fc..c82621536f01b488b4d4cd0dccd1d2f7dddc3175). 
My only worry is that if they change the error format then it might break, but I'm over analyzing this thing, so let's just keep it like this

Reply:
sgtm 
only thing that worth considering later if desired is if there are any side effects in case we match the last assertion 
(when all the other branches didnt meet)